projects/planner/src/app/learn/learn.component.ts

```diff
@@ -243,7 +226,7 @@ export class LearnComponent implements OnInit {
         if (existing) {
             return existing;
         }
-        const category = this.findCategory(this.selectedTopic, categoryName);
+        const category = this.selectedTopic.getCategoryByNameOrEmpty(categoryName);
```

Comment:
replace categoryName by category, since it is available for all calls.
Then you dont need to search for the category